packages/axelar-wasm-std-derive/src/lib.rs

```diff
@@ -282,3 +282,71 @@ fn match_unit_variant(event_enum: &Ident, variant_name: &Ident) -> TokenStream2
         #event_enum::#variant_name => cosmwasm_std::Event::new(#event_name)
     }
 }
+
+#[proc_macro_attribute]
+pub fn migrate_from_version(input: TokenStream, item: TokenStream) -> TokenStream {
+    let base_version_req = syn::parse_macro_input!(input as syn::LitStr).value();
+    let annotated_fn = syn::parse_macro_input!(item as syn::ItemFn);
+
+    let fn_name = &annotated_fn.sig.ident;
+    let fn_inputs = &annotated_fn.sig.inputs;
+    let fn_output = &annotated_fn.sig.output;
+    let fn_block = &annotated_fn.block;
+
+    if fn_name != "migrate" {
+        return syn::Error::new(
+            fn_name.span(),
+            "#[migrate_from_version] can only be applied to a 'migrate' function",
+        )
+        .to_compile_error()
+        .into();
+    }
+
+    let deps = match deps_ident(&annotated_fn.sig) {
```

Comment:
If `DepsMut` being the first param is so important, why not also check the other params to make sure the migrate function matches the right signature?